### PR TITLE
fix: fix SwiftPackageManager dependencies mapping on Xcode 13

### PR DESCRIPTION
**Pull request by danyf90**
Resolves https://github.com/tuist/tuist/issues/YYY
Request for comments document (if applies):

### Short description 📝

One field is changed in swift 5.5, make fetch work on both versions

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Models/SwiftPackageManagerWorkspaceState.swift [hunk 1]
```diff
@@ -24,7 +24,9 @@ struct SwiftPackageManagerWorkspaceState: Decodable, Equatable {
         let name: String
         /// The king of the dependency (either local or remote)
         let kind: String
-        /// The path of the local dependency
+        /// The path of the local dependency, no longer available since Swift 5.5
```
Comment (laxmorek) [inline Sources/TuistDependencies/SwiftPackageManager/Models/SwiftPackageManagerWorkspaceState.swift:27]: swift version or tools version? 🤔

File: Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift [hunk 2]
```diff
@@ -85,7 +85,7 @@ public final class SwiftPackageManagerGraphGenerator: SwiftPackageManagerGraphGe
             case "remote":
                 packageFolder = checkoutsFolder.appending(component: dependency.subpath)
             case "local":
-                guard let path = dependency.packageRef.path else {
+                guard let path = dependency.packageRef.path ?? dependency.packageRef.location else {
```
Comment (laxmorek) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:88]: maybe you could leave a message why `path` fallbacks to `location` 

it is not obvious on the first glance

**Review by laxmorek — APPROVED** [review]

File: Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift (hunk 2 above)
Comment (laxmorek) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:89]: you could consider creating a new computable variable inside `packageRef` that implements this fallback

Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:89]: I would keep that type just as a model (as the PackageInfo), and keep the logic here

**Review by fortmarek — APPROVED** [review]

**Pull request merged**